source/testing/index.md

```diff
@@ -1,79 +1,149 @@
 Testing is a core part of the Ember framework and its development cycle.
 
 Let's assume you are writing an Ember application which will serve as a blog.
-This application would likely include models such as `user` and `post`. It would
-also include interactions such as _login_ and _create post_. Let's finally
-assume that you would like to have [automated tests] in place for your application.
+This application would likely include models such as `user` and `post`.
+It would also include interactions such as _login_ and _create post_.
+Let's finally assume that you would like to have [automated tests] in place for your application.
 
-There are three different classifications of tests that you will need:
-**Acceptance**, **Unit**, and **Integration**.
+There are four different types of test setups you can choose from depending on what you will need to test.
+Let's have a look at those test types and when each of them is the most useful.
 
-### Acceptance Tests
+### Application Tests
 
-Acceptance tests are used to test user interaction and application flow.  The tests interact
-with the application in the same ways that a user would, by doing things like filling out
-form fields and clicking buttons.  Acceptance tests ensure that the features within
-a project are basically functional, and are valuable in ensuring the core features of a
-project have not regressed, and that the project's goals are being met.
+Some tests will require you to test user interaction and application flow.
+In these kind of tests you would like to be able to interact with the application in the same ways that a user would,
+by doing things like filling out form fields and clicking buttons.
+This will ensure that the features within a project are basically functional,
+and ensure that the core features of a project have not regressed, and that the project's goals are being met.
 
-In the example scenario above, some acceptance tests one might write are:
+In the example scenario above, some of these tests you might write are:
 
 * A user is able to log in via the login form.
 * A user is able to create a blog post.
 * After saving a new post successfully, a user is then shown the list of prior posts.
 * A visitor does not have access to the admin panel.
 
-### Unit Tests
+For tests that require a full application setup we can use the `setupApplicationTest` helper.
+You can read more about how to create these kind of tests in the [Application tests] section.
 
-Unit tests are used to test isolated chunks of functionality, or "units".
-They can be written against any isolated application logic.
+### Rendering Tests
 
-Some specific examples of units tests are:
+Some tests serve as a middle ground between those tests, which require a full application setup and those,
+which interact with specific code algorithms on a micro level.
```

Comment:
What about when you need to test the middle ground between a full application setup and unit tests of specific functions? Consider rendering tests.